test/mongodb.test.js

```diff
@@ -4,6 +4,36 @@ var should = require('./init.js');
 
 var Superhero, User, Post, PostWithStringId, db;
 
+describe('lazyConnect', function() {
+  it('should skip connect phase (lazyConnect = true)', function(done) {
+    var ds = getDataSource({
+      host: '127.0.0.1',
+      port: 4,
+      lazyConnect: true,
+    });
+    var errTimeout = setTimeout(function() {
+      done();
+    }, 5000);
+
+    ds.on('error', function(err) {
+      clearTimeout(errTimeout);
```

Comment:
I think you should also fail the test in this case, because the error was not supposed to be triggered, is my understanding correct?

```js
clearTimeout(errTimeout);
done(err);
```